File: mycollab-services-community/src/main/java/com/esofthead/mycollab/community/module/common/service/ibatis/TagServiceImpl.java
/**
 * This file is part of mycollab-services-community.
 *
 * mycollab-services-community is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-services-community is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-services-community.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.community.module.common.service.ibatis;

import com.esofthead.mycollab.common.domain.AggregateTag;
import com.esofthead.mycollab.common.domain.Tag;
import com.esofthead.mycollab.common.domain.TagExample;
import com.esofthead.mycollab.common.service.TagService;
import com.esofthead.mycollab.core.cache.CacheKey;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * @author MyCollab Ltd
 * @since 5.3.4
 */
@Service
public class TagServiceImpl implements TagService {
    @Override
    public List<Tag> findTags(String type, String typeId, @CacheKey Integer accountId) {
        return null;
    }

    @Override
    public List<Tag> findTagsInAccount(String name, String[] types, @CacheKey Integer accountId) {
        return null;
    }

    @Override
    public List<AggregateTag> findTagsInProject(Integer projectId, @CacheKey Integer accountId) {
        return null;
    }

    @Override
    public int deleteByExample(TagExample example) {
        return 0;
    }

    @Override
    public Integer saveWithSession(@CacheKey Tag record, String username) {
        return null;
    }

    @Override
    public Integer updateWithSession(@CacheKey Tag record, String username) {
        return null;
    }

    @Override
    public Integer updateSelectiveWithSession(@CacheKey Tag record, String username) {
        return null;
    }

    @Override
    public void massUpdateWithSession(Tag record, List<Integer> primaryKeys, @CacheKey Integer accountId) {

    }

    @Override
    public Tag findByPrimaryKey(Integer primaryKey, @CacheKey Integer sAccountId) {
        return null;
    }

    @Override
    public void removeWithSession(Tag item, String username, @CacheKey Integer sAccountId) {

    }

    @Override
    public void massRemoveWithSession(List<Tag> items, String username, @CacheKey Integer sAccountId) {

    }
}

File: mycollab-services-community/src/main/java/com/mycollab/community/module/common/service/impl/TagServiceImpl.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.community.module.common.service.impl

import com.mycollab.common.domain.AggregateTag
import com.mycollab.common.domain.Tag
import com.mycollab.common.domain.TagExample
import com.mycollab.common.service.TagService
import com.mycollab.core.cache.CacheKey
import org.springframework.stereotype.Service

/**
 * @author MyCollab Ltd
 * @since 5.3.4
 */
@Service
class TagServiceImpl : TagService {
    override fun findTags(type: String, typeId: String, @CacheKey accountId: Int?): List<Tag>? {
        return null
    }

    override fun findTagsInAccount(name: String, types: Array<String>, @CacheKey accountId: Int?): List<Tag>? {
        return null
    }

    override fun findTagsInProject(projectId: Int, accountId: Int): List<AggregateTag>? {
        return null
    }

    override fun deleteByExample(example: TagExample): Int? {
        return 0
    }

    override fun saveWithSession(@CacheKey record: Tag, username: String?): Int {
        return -1
    }

    override fun updateWithSession(@CacheKey record: Tag, username: String?): Int {
        return -1
    }

    override fun updateSelectiveWithSession(@CacheKey record: Tag, username: String?): Int? {
        return null
    }

    override fun massUpdateWithSession(record: Tag, primaryKeys: List<Int>, @CacheKey accountId: Int?) {

    }

    override fun findByPrimaryKey(primaryKey: Int, sAccountId: Int): Tag? {
        return null;
    }

    override fun removeWithSession(item: Tag, username: String?, sAccountId: Int) {

    }

    override fun massRemoveWithSession(items: List<Tag>, username: String?, sAccountId: Int) {

    }
}
